### PR TITLE
Add delay and throttle for reassigning buckets

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java [hunk 1]
```diff
@@ -537,6 +541,9 @@ public void createBuckets(long distributionId, int bucketCount)
                 if (gracePeriod && (nanosSince(startTime).compareTo(startupGracePeriod) < 0)) {
                     throw new PrestoException(SERVER_STARTING_UP, "Cannot reassign buckets while server is starting");
                 }
+                if (gracePeriod) {
+                    assignmentLimiter.checkAssignFrom(nodeId);
```
Comment (dain) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java]: Why put this behind the grace period check if the defaults are `0 minutes`

**Comment by dain** [comment]
This looks good, but you have compile errors

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java (hunk 1 above)
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java]: That variable means whether or not the caller wants to use the grace period check. Currently only the tests use it to skip it. It's more needed for the node assignments. I could refactor this code to remove it.

File: presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestAssignmentLimiter.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        limiter.checkAssignFrom("A");
+
+        // 4:30
+        ticker.increment(5, MINUTES);
+        limiter.checkAssignFrom("A");
+        limiter.checkAssignFrom("B");
+        limiter.checkAssignFrom("C");
+    }
+
+    private static void assertCheck(AssignmentLimiter limiter, String node, ErrorCodeSupplier expected)
```
Comment (dain) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestAssignmentLimiter.java]: name this `assertCheckFails`

**Comment by dain** [comment]
Looks good

**Pull request closed without merging**